Formula/q/qbittorrent-cli.rb

```diff
@@ -0,0 +1,20 @@
+class QbittorrentCli < Formula
+  desc "Command-line interface for qBittorrent written in Go"
+  homepage "https://github.com/ludviglundgren/qbittorrent-cli"
+  url "https://github.com/ludviglundgren/qbittorrent-cli/archive/refs/tags/v2.0.0.tar.gz"
+  sha256 "91969f22bb167f99091a1ff8f4dbfe61cb1b592cb000453859ca9173ecbb8f10"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "cmd/qbt/main.go"
+  end
+
+  test do
+    version = "dev"
+    commit = "none"
+    date = "unknown"
+    assert_equal "qbt - qbitttorrent cli\nVersion: #{version}\nCommit: #{commit}\nDate: #{date}", shell_output("#{bin}/qbittorrent-cli version").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I understand completely...but... 😄 
This software is a command line client for controlling a bittorrent GUI (which does not have to run on the same system). Do you have specific advice for that? I was trying to come up with a test that does not require other formulas to be installed.

Reply:
@chenrui333 I added another test simulating a qbittorrent API endpoint in a basic manner. Would that be satisfactory?